io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/writer/AbstractTableInstanceWriter.java

```diff
@@ -225,20 +234,40 @@ private boolean shouldBeDisplayed(Object obj) {
 	}
 
 	private void addProperty(List<String> headerRow, Map<String, Object> row, Object property,
-			String propertyTypeName) {
+			String propertyTypeName, IOReporter reporter) {
 		if (!headerRow.contains(propertyTypeName)) {
 			headerRow.add(propertyTypeName);
 		}
+
+		if (property instanceof DefaultGeometryProperty<?>) {
+			// Property is an instance of DefaultGeometryProperty<T>
+			DefaultGeometryProperty<?> geometryProperty = (DefaultGeometryProperty<?>) property;
+
+			Geometry geometry = geometryProperty.getGeometry();
+
+			// correct winding order as per right-hand rule, i.e.,
+			// exterior rings are counterclockwise, and holes are
+			// clockwise.
+			setWindingOrder(EnumWindingOrderTypes.counterClockwise);
+			property = unifyGeometry(geometry, reporter,
+					geometryProperty.getCRSDefinition().getCRS());
+
+			if (geometry instanceof GeometryCollection
+					&& ((GeometryCollection) geometry).getNumGeometries() == 1) {
+				property = geometryProperty.getGeometry().getGeometryN(0);
```

Comment:
Like this the CRS information gets lost.

Reply:
@stempler what it was done here was just after taking the geometry from the property, so after `(GeometryProperty<?>) property`. If you tell me that I need to check if the `property is an instanceof GeometryCollection` then that condition it will never be true. 
In the InstanceToJson is another story, already in the method comes a parameter which is a `GeometryProperty<?>`.

Reply:
I'm afraid I don't understand the context of your comment.

> If you tell me that I need to check if the property is an instanceof GeometryCollection then that condition it will never be true

If you refer to my comment [here](https://github.com/halestudio/hale/pull/1173#discussion_r1613705904)

1. this comment was on another place in the code where the JTS geometry is handled, not the geometry property
2. I did not tell you anything, I just asked if there was a specific reason for the change to check for some `GeometryCollection` subclasses instead of the base class